src/circuit/gadget/ecc.rs

```diff
@@ -376,3 +378,168 @@ impl<C: CurveAffine, EccChip: EccInstructions<C> + Clone + Debug + Eq> FixedPoin
             })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::constants;
+    use ff::Field;
+    use group::{Curve, Group};
+    use halo2::{
+        arithmetic::{CurveAffine, FieldExt},
+        circuit::{layouter::SingleChipLayouter, Layouter},
+        dev::MockProver,
+        pasta::pallas,
+        plonk::{Assignment, Circuit, ConstraintSystem, Error},
+    };
+
+    use super::chip::{EccChip, EccConfig, OrchardFixedBases, OrchardFixedBasesShort};
+
+    struct MyCircuit<C: CurveAffine> {
+        _marker: std::marker::PhantomData<C>,
+    }
+
+    #[allow(non_snake_case)]
+    impl<C: CurveAffine> Circuit<C::Base> for MyCircuit<C> {
+        type Config = EccConfig;
+
+        fn configure(meta: &mut ConstraintSystem<C::Base>) -> Self::Config {
+            let bits = meta.advice_column();
+            let P = (meta.advice_column(), meta.advice_column());
+            let lambda = (meta.advice_column(), meta.advice_column());
+            let extras = [
+                meta.advice_column(),
+                meta.advice_column(),
+                meta.advice_column(),
+                meta.advice_column(),
+                meta.advice_column(),
+            ];
```

Comment:
Please add documentation of how the extras are used. Perhaps have either constants specifying the offsets into `extras`, or references to them (if that doesn't cause too many borrow checker problems).

Reply:
This will change in any case now that we've optimized out A, B, C, and D.

Reply:
I've defined a separate `Config` for each operation (e.g. `add::Config, mul_fixed::Config`), each with an `impl From<EccConfig>` showing how it is derived from the top-level config.